lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -230,20 +238,24 @@ def acceptable?(scopes)
     # We keep a volatile copy of the raw refresh token for initial communication
     # The stored refresh_token may be mapped and not available in cleartext.
     def plaintext_refresh_token
-      if perform_secret_hashing?
-        @raw_refresh_token
+      if secret_strategy.allows_restoring_secrets?
+        secret_strategy.restore_secret(refresh_token)
```

Comment:
I like the idea of external secret strategies. The only thing that we need to think about: Currently we using the same column for `find_by_plaintext_token` and `restore_secret`. I don't think that it will work

If we want to use these strategies for encryption, then we somehow must keep hash of token to be able to index it for `find_by_plaintext_token` and in the same time we must keep ciphertext for `restore_secret`.

Reply:
Makes sense to me too

Reply:
I see. Let me think about how we could integrate that into the concern. We would need to find the token value by hash and then pass that instance to `restore_secret` so you would have access to another column

Reply:
I believe that this is resolved now with `store_secret` and `restore_secret` receiving the model instance in order to modify that value however it likes. This could mean storing a hash or doing something else with it.